Content.Server/Magic/MagicSystem.cs

```diff
@@ -173,6 +178,27 @@ private void OnProjectileSpell(ProjectileSpellEvent ev)
         }
     }
 
+    private void OnChangeComponentsSpell(ChangeComponentsSpellEvent ev)
+    {
+        foreach (var toRemove in ev.ToRemove)
+        {
+            if (_compFact.TryGetRegistration(toRemove, out var registration))
+                RemComp(ev.Target, registration.Type);
+        }
+
+        foreach (var (name, data) in ev.ToAdd)
+        {
+            if (HasComp(ev.Target, data.Component.GetType()))
+                continue;
+
+            var component = (Component) _compFact.GetComponent(name);
+            component.Owner = ev.Target;
+            var temp = (object) component;
+            _seriMan.CopyTo(data.Component, ref temp);
+            EntityManager.AddComponent(ev.Target, (Component) temp!);
+        }
```

Comment:
One very big issue with this: The components are added one at a time. This includes their initialization order, so each one will receive startup events one at a time.

Reply:
AFAIK there is currently no support for adding several components simultaneously after an entity has been initialized, and I imagine that for most uses this wouldn't be a problem.

Eventually it'd be nice to support adding mutually dependent components but that requires engine changes & would also require updating things like `AddComponentSpecial` and other yaml defined component-adding interactions, so I' rather just open an engine issue for this & fix that separately than delay this PR.

Reply:
sgtm